include/circt/Dialect/HW/Passes.td

```diff
@@ -44,6 +44,20 @@ def FlattenIO : Pass<"hw-flatten-io", "mlir::ModuleOp"> {
   ];
 }
 
+def FlattenModules : Pass<"hw-flatten-modules", "mlir::ModuleOp"> {
+  let summary = "Eagerly inline private modules";
+  let description = [{
+    This pass eagerly inlines private HW modules into their instantiation sites.
+    After outlining combinational logic and registers into arcs, module bodies
+    become fairly lightweight. Since arc definitions now fulfill the purpose of
+    code reuse by allowing a single definition to be called multiple times, the
+    module hierarchy degenerates into a purely cosmetic construct. At that point
+    it is beneficial to fully flatten the module hierarchy to simplify further
+    analysis and optimization of state transfer arcs.
```

Comment:
Could you rephrase these sentences a bit since it's HW pass now?

Reply:
yeah ofc!